tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/screens/photo/PhotoViewModel.kt

```diff
@@ -29,19 +31,21 @@ class PhotoViewModel @Inject constructor(
     }
 
     private suspend fun updatePhoto() {
-        when (val res = repository.getPhotosUrl("\"/BAND/Photos/Albums/Photo_March\"")) {
-            is Resource.Error -> {
-                _state.update { state ->
-                    state.copy(isSuccess = false, error = res.error)
-                }
-                Log.e("TAG", res.error)
-            }
-            is Resource.Data -> {
-                Log.d("href", res.data.toString())
-                _state.update { state ->
-                    state.copy(photos = res.data.map { Photo(it) }, isSuccess = true)
-                }
-            }
+       repository.getPhotosUrl("\"${BuildConfig.folderPathPhotoSynology}\"").collect { result->
+           when(result) {
+               is Resource.Error -> {
```

Comment:
Why you use `Resource` here instead of `Either` ?

Reply:
This is a wrapper for domain-level data. It seemed to me that it would be more convenient